src/cls/rbd/cls_rbd_types.h

```diff
@@ -732,6 +732,153 @@ struct SnapshotInfo {
 };
 WRITE_CLASS_ENCODER(SnapshotInfo);
 
+enum GroupSnapshotNamespaceType {
```

Comment:
We want to include a corresponding enum in `src/include/rbd/librbd.hpp` and `src/include/rbd/librbd.h` . We can then expose this using the new APIs to get information about a group snapshot or list all group snapshots.

e.g., in `src/include/rbd/librbd.hpp`
```

typedef enum {
  RBD_GROUP_SNAP_NAMESPACE_TYPE_USER   = 0,
  RBD_GROUP_SNAP_NAMESPACE_TYPE_MIRROR = 1
} rbd_group_snap_namespace_type_t;
```

See Ilya's comment https://github.com/ceph/ceph/pull/57797/files#r1663974868 for more details.

Reply:
Also see https://github.com/ceph/ceph/pull/57797#discussion_r1663997770